snap/snapcraft.yaml

```diff
@@ -543,30 +546,63 @@ parts:
       # Do the actual pull task
       craftctl default
       # Settings:
-      # Grade: stable/devel
-      GRADE=stable
-      # Package release number (integer)
-      PACKAGERELEASE=1
+      # Force channel: auto/devel/edge/stable
+      CHANNEL=stable
+      # Force package release number (integer) or "auto"
+      PACKAGERELEASE=auto
+      # Time stamp of Snap build in Snap Store
+      snapbuilddatehuman=`curl -s -H 'Snap-Device-Series: 16' https://api.snapcraft.io/v2/snaps/info/ps-printer-app | jq -r '."channel-map" | .[] | select(.channel.name == "edge") | select(.channel.architecture == "amd64") | ."created-at"'`
+      snapbuilddate=`date +%s --date=$snapbuilddatehuman`
+      if [ -z "$snapbuilddate" ]; then
+          snapbuilddate=0
+      fi
+      # Time stamp of the last GIT commit of the snapping repository
+      pushd $CRAFT_PROJECT_DIR
+      gitcommitdate=`git log -1 --date=unix | grep Date: | perl -p -e 's/Date:\s*//'`
+      popd
+      if [ -z "$gitcommitdate" ]; then
+          gitcommitdate=0
+      fi
+      # Previous stable and development version
+      prevstable="$(curl -s -H 'Snap-Device-Series: 16' https://api.snapcraft.io/v2/snaps/info/ps-printer-app | jq -r '."channel-map" | .[] | select(.channel.name == "stable") | select(.channel.architecture == "'$CRAFT_TARGET_ARCH'") | .version')"
+      if [ -z "$prevstable" ]; then
+          prevstable=0
+      fi
+      prevdevel="$(curl -s -H 'Snap-Device-Series: 16' https://api.snapcraft.io/v2/snaps/info/ps-printer-app | jq -r '."channel-map" | .[] | select(.channel.name == "edge") | select(.channel.architecture == "'$CRAFT_TARGET_ARCH'") | .version')"
+      if [ -z "$prevdevel" ]; then
+          prevdevel=0
+      fi
+      # Previous version in general
+      dpkg --compare-versions "$prevdevel" lt "$prevstable" && prevversion=$prevstable || prevversion=$prevdevel
       # Current upstream version of foomatic-db
-      #
-      # As foomatic-db is simply a collection of printer data and not
-      # some software which is under continuous development and
-      # milestones of the development turn into releases, it is simply
-      # versioned by the date (a cron job on OpenPrinting "releases"
-      # it every day.
-      #
-      # As we want to download it from GIT and not grab a daily
-      # tarball for sake of Snap automation we simply take the date of
-      # the day after the day of the execution of the pull task of
-      # this part as the version number (this is the version in which
-      # the current GIT state would land).
```

Comment:
Please do not remove this comment, put it to the beginning of the version automation code.